app/src/androidTest/java/ch/epfl/sweng/SDP/auth/LoginActivityTest.java

```diff
@@ -37,51 +47,99 @@
+        mockIntent = Mockito.mock(Intent.class);
+        Mockito.when(mockIdpResponse.getEmail()).thenReturn("testEmail");
+        Mockito.when(mockIntent.getParcelableExtra(ExtraConstants.IDP_RESPONSE))
+                .thenReturn(mockIdpResponse);
     }
 
     @Test
     public void testFailedLoginNullResponse() {
-        Intent mockIntent = Mockito.mock(Intent.class);
         Mockito.when(mockIntent.getParcelableExtra(ExtraConstants.IDP_RESPONSE))
                 .thenReturn(null);
-        loginActivity.onActivityResult(42, 0, mockIntent);
+        activityRule.getActivity().onActivityResult(42, 0, mockIntent);
         assertThat(loginActivity.isFinishing(), is(true));
     }
 
     @Test
     public void testSuccessfulLoginNewUser() {
-        Intent mockIntent = Mockito.mock(Intent.class);
-        Mockito.when(mockIntent.getParcelableExtra(ExtraConstants.IDP_RESPONSE))
-                .thenReturn(mockIdpResponse);
         Mockito.when(mockIdpResponse.isNewUser()).thenReturn(true);
-        loginActivity.onActivityResult(42, -1, mockIntent);
+        activityRule.getActivity().onActivityResult(42, -1, mockIntent);
         assertTrue(loginActivity.isFinishing());
         Activity accountCreationActivity = getInstrumentation()
                 .waitForMonitorWithTimeout(monitor, 5000);
         assertThat(accountCreationActivity, is(not(nullValue())));
     }
 
     @Test
-    public void testCloneAccountFromFirebase() {
-        DataSnapshot mockSnapshot = Mockito.mock(DataSnapshot.class);
-        HashMap<String, HashMap<String, Object>> userEntry = new HashMap<>();
-        Mockito.when(mockSnapshot.getValue())
-                .thenReturn(userEntry);
-        assertThat(mockSnapshot.getValue(), CoreMatchers.<Object>is(userEntry));
+    public void testSuccessfulLoginExistingUser() {
+        Mockito.when(mockIdpResponse.isNewUser()).thenReturn(false);
+        activityRule.getActivity().onActivityResult(42, -1, mockIntent);
+        SystemClock.sleep(3000);
+        assertTrue(loginActivity.isFinishing());
+        Activity homeActivity = getInstrumentation()
+                .waitForMonitorWithTimeout(homeMonitor, 5000);
+        Assert.assertNotNull(homeActivity);
+    }
+
+    @Test
+    public void testSuccessfulLoginWithoutAccount() {
+        Mockito.when(mockIdpResponse.getEmail()).thenReturn("weirdEmail");
+        Mockito.when(mockIdpResponse.isNewUser()).thenReturn(false);
+        activityRule.getActivity().onActivityResult(42, -1, mockIntent);
+        Activity accountCreationActivity = getInstrumentation()
+                .waitForMonitorWithTimeout(monitor, 5000);
+        Assert.assertNotNull(accountCreationActivity);
+    }
+
+    @Test
+    public void testFailedLoginNoNetwork() {
+        failedSignInHelper(R.string.no_internet, ErrorCodes.NO_NETWORK);
+    }
+
+    @Test
+    public void testFailedLoginUnknownError() {
+        failedSignInHelper(R.string.unknown_error, 1234567891);
```

Comment:
What is this number here?

Reply:
This is any random number that doesn't belong to an ErrorCode. This is to test how the class reacts to an unknown ErrorCode.

Reply:
Okey, maybe comment this or add define a variable named randomNumber to make it clearer